.travis/build-macos.sh

```diff
@@ -4,6 +4,68 @@ if echo $TRAVIS_TAG | grep ^v; then BUILD_TYPE=RelWithDebInfo; else BUILD_TYPE=D
 
 mkdir build
 cd build
-cmake -DCMAKE_OSX_DEPLOYMENT_TARGET=10.7 -DCMAKE_BUILD_TYPE=$BUILD_TYPE ..
-make VERBOSE=1
+cmake -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 -DCMAKE_BUILD_TYPE=$BUILD_TYPE ..
+make -j 2 VERBOSE=1
 make test_solvespace
+
+ls bin
+
+app="bin/SolveSpace.app"
+dmg="bin/SolveSpace.dmg"
+bundle_id="com.solvespace.solvespace"
+
+# get the signing certificate (this is the Developer ID:Application: Your Name, exported to a p12 file, then converted to base64, e.g.: cat ~/Desktop/certificate.p12 | base64 | pbcopy)
+echo $MACOS_CERTIFICATE_P12 | base64 --decode > certificate.p12
+
+# create a keychain
+security create-keychain -p secret build.keychain
+security default-keychain -s build.keychain
+security unlock-keychain -p secret build.keychain
+
+# import the key
+security import certificate.p12 -k build.keychain -P $MACOS_CERTIFICATE_PASSWORD -T /usr/bin/codesign
+
+security set-key-partition-list -S apple-tool:,apple: -s -k secret build.keychain
+#security set-keychain-settings -t 3600 -u build.keychain
+
+# check if all is good
+security find-identity -v
+
+# sign the .app
+codesign -s "${MACOS_DEVELOPER_ID}" --timestamp --options runtime -f --deep "${app}"
+
+# create the .dmg from the signed .app
+hdiutil create -srcfolder "${app}" "${dmg}"
+
+# sign the .dmg
+codesign -s "${MACOS_DEVELOPER_ID}" --timestamp --options runtime -f "${dmg}"
+
+# notarize and store request uuid in variable
+notarize_uuid=$(xcrun altool --notarize-app --primary-bundle-id "${bundle_id}" --username "${MACOS_APPSTORE_USERNAME}" --password "${MACOS_APPSTORE_APP_PASSWORD}" --file "${dmg}" 2>&1 | grep RequestUUID | awk '{print $3'})
```

Comment:
I'm a bit confused by this. Does XCode offer no way to block until notarization is finished?

Reply:
You are correct =) There is no way to block indeed, a very anti-CI approach. The voodoo is necessary unfortunately...